frontend/src/components/SystemSubmitDrawer/index.tsx

```diff
@@ -436,24 +436,46 @@ export function SystemSubmitDrawer(props: Props) {
       });
   }
 
-  function getDataFileValidator(isRequired: boolean, fieldName: string) {
+  /**
+   *
+   * @param isRequired If the field is not required, the validator will always pass.
+   * @param fieldName For display purpose only. Will be shown in warnings.
+   * @param singleFile If validating multiple files, will check for each file's name
+   * @returns whether the form is valid nor not
+   */
+  function getDataFileValidator(
+    isRequired: boolean,
+    fieldName: string,
+    singleFile: boolean
+  ) {
     if (!isRequired) return () => Promise.resolve();
     else
       return (_: unknown, value: DataFileValue) => {
-        let valid;
-        valid =
-          value &&
-          value.fileList &&
-          value.fileType &&
-          value.fileList.length > 0;
-        if (value.sysNames) {
-          for (const name of Object.values(value.sysNames)) {
-            valid = valid && name;
+        if (
+          !value ||
+          !value.fileList ||
+          !value.fileType ||
+          value.fileList.length === 0
+        ) {
+          return Promise.reject(`'${fieldName}' and file type are required`);
+        }
+
+        if (!singleFile) {
+          if (!value.sysNames || !value.fileList) {
```

Comment:
It seems that this error message is incorrect. As you mentioned in the description, `value.sysNames` is `undefined` when it is initialized so this is an expected scenario. Maybe we should just prompt the user to fill in the system name instead of refreshing the page?

Also, I think this function now has different behavior depending on the value of `singleFile`. It seems to me that the difference is really related to whether this function is used for the dataset form item or for the system output form item. I think it will make things clearer if we have two separate validator functions. One for the dataset and another one for system output. The system output one can call the dataset one because it seems to have some overlap. Or we can have a third helper function that is used by both functions. WDYT?

Also, `FileSelect` is a custom `Form.Item` we implemented so you can modify how it behaves. I think you can assign a more appropriate default value for `sysNames` which may help you to avoid some of the issue. But this may not be easy to do so I'm ok with having two separate validator functions.